### PR TITLE
Add custom actions for Python and Node.js

### PR DESCRIPTION
**Pull request by kalrish**
I propose two new [custom actions](https://docs.github.com/en/actions/creating-actions/about-custom-actions):

- `setup-python-Poetry`, to set up Python with the [action `setup-python` from GitHub](https://github.com/marketplace/actions/setup-python) and install dependencies with [Poetry](https://python-poetry.org/); and
- `setup-node-npm`, to set up Node.js with the [action `setup-node` from GitHub](https://github.com/marketplace/actions/setup-node-js-environment) and install dependencies with npm.

Both of these actions would be used by reusable workflows hosted in this repository and possibly by external workflows.

### CODE REVIEW

File: .github/actions/setup-python-Poetry/action.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      with:
+        # FIXME: Enable cache
+        # If uncommented, the action fails with the following error message:
+        #
+        # > Unable to locate executable file: poetry.
+        #
+        # This error message suggests that the action needs Poetry to restore its cache.
+        # How can Poetry be installed before this action, which sets up Python?
+        #
+        # cache: poetry
```
Comment (EduardSchwarzkopf) [inline .github/actions/setup-python-Poetry/action.yaml:50]: still relevant?

**Review by EduardSchwarzkopf — COMMENTED** [review]
check the one comment from the big FIX ME part

File: .github/actions/setup-python-Poetry/action.yaml (hunk 1 above)
Comment (kalrish) [reply .github/actions/setup-python-Poetry/action.yaml:50]: Actually, the [relevant section of the documentation for the action `setup-python`](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages) provides the following example to cache Poetry dependencies:

```
steps:
- uses: actions/checkout@v3
- name: Install poetry
  run: pipx install poetry
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'poetry'
- run: poetry install
- run: poetry run pytest
```

I had never heard of `pipx`. I smell a rabbit hole. I'm going in…

Comment (kalrish) [reply .github/actions/setup-python-Poetry/action.yaml:50]: The [images for GitHub Actions runners](https://github.com/actions/runner-images#available-images) contain (different versions of) Python and pip and also [pipx](https://pypa.github.io/pipx/). I'll add a commit to install Poetry before calling `actions/setup-python` and enable the cache.

**Review by EduardSchwarzkopf — APPROVED** [review]
sounds good to me

**Pull request merged**